src/bot/commands/campaign.ts

```diff
@@ -126,30 +126,37 @@ const command: ICommand = {
               where: { roomId: room.id },
               relations: ['characters'],
             });
-            console.log(result);
+
             if (result.length > 0) {
               const campaign = result[0];
 
-              const currentMap = JSON.parse(campaign.dungeon);
-              const myPos = JSON.parse(matchedChar.position);
-              for (let i = 0; i < currentMap.world.length; i += 1) {
-                for (let j = 0; j < currentMap.world[i].length; j += 1) {
-                  if (currentMap.world[i][j] === 5) {
-                    myPos.x = j;
-                    myPos.y = i;
-                    break;
-                  }
+              // eslint-disable-next-line consistent-return
+              let alreadyJoined = false;
+              campaign.characters.forEach((char) => {
+                if (char.name === matchedChar.name) {
+                  alreadyJoined = true;
+                }
+              });
```

Comment:
You can improve the efficiency of this part by using `Array.prototype.some`, which also short-circuits the loop (while this doesn't).
```suggestion
              const alreadyJoined = campaign.characters.some((char) => char.name === matchedChar.name);
```

Reply:
I already did a similar thing here: 4f82e0cbea6934c11428c7648e163dacb70b1779

Reply:
Similarly, that one doesn't short-circuit :) but it is definitely cleaner